src/main/kotlin/org/rust/ide/annotator/RsSyntaxErrorsAnnotator.kt

```diff
@@ -165,6 +170,27 @@ private fun checkConstant(holder: AnnotationHolder, const: RsConstant) {
             require(const.expr, holder, "$title must have a value", const)
         }
     }
+    checkConstantType(holder, const)
+}
+
+private fun checkConstantType(holder: AnnotationHolder, element: RsConstant) {
+    if (element.colon == null && element.typeReference == null) {
+        val typeText = if (element.static != null) { "static" } else { "const" }
+        val message = "Missing type for `$typeText` item"
+
+        val annotation = holder.newAnnotation(HighlightSeverity.ERROR, message)
+            .range(element.textRange)
+
+        val expr = element.expr
+        if (expr != null) {
+            val type = expr.type
+            if (!type.containsTyOfClass(TyUnknown::class.java)) {
```

Comment:
Maybe it make sense to suggest this quick fix even final type is not fully known by the plugin?
A user should insert it anyway and in case of partially known type the plugin may insert significant part of the correct type (especially, if type is complex). Also, if it will be completely wrong, a user just replace it since we use template builder
What do you think?

Reply:
When the type is `TyUnknown`, it's probably faster to type `:` than to invoke the quick fix and then delete the unknown type. But you're right that the current check might be too strict. So just check if it isn't `TyUnknown` and otherwise offer it? That sounds reasonable.

Reply:
> When the type is TyUnknown, it's probably faster to type : than to invoke the quick fix and then delete the unknown type.

Probably, it's indeed faster. But an additional quick fix doesn't interfere with manual code modification at all. Also, I don't know other quick fixes/intentions that can interfere in a situation when constant doesn't have explicit type. So I'd prefer just always add this quick fix (it also simplifies code a little bit) if you don't have strong objections

Reply:
Fair enough, it's better to be consistent! I removed the check for `TyUnknown` :+1: